in-progress/12487-prd-fast-sync.md

```diff
@@ -0,0 +1,57 @@
+# Fast-Sync Project Requirements Document
+
+- Owners: @spalladino, @aminsammara
+- Approvers:
+  - @AndreOxski
+  - @charlielye
+  - @PhilWindle
+  - @joshcrites
+
+## Background
+
+Syncing an L2 node requires crawling L1 since L2 genesis, downloading events and blobs, and updating world-state for every update. This is time-consuming, and proportional to the length of the chain.
+
+## Desired User Flow
+
+Node operators should be able to sync the chain in nearly constant time by downloading a recent snapshot, and syncing only the most recent blocks. Operators should be able to do this by specifying a snapshot URL to sync from.
+
+This flow is available in most Ethereum clients.
+
+### Node operators
+
+Users SHOULD be able to specify which method to use to sync their clients. Methods are either `fast` (snapshots) or `full` (for syncing from the chain). Default is `fast`. For example:
+`aztec start --node --publisher --sync-method fast`
+
+Users SHOULD also be able to specify the location for the snapshot. Protocols supported are ipfs and https. For example:
+`aztec start --node --publisher --sync-method fast --snapshot-url ipfs://...`
+
+Users SHOULD also be able to specify the location for a snapshot index, which is a machine-readable file that lists the latest snapshots. Protocols supported https only. For example:
+`aztec start --node --publisher --sync-method fast --snapshot-index https://...`
```

Comment:
So the idea is the node software would read the index and automatically go and find that snapshot?

Reply:
Yep. In my mind the index is a list of snapshots that gets updated, sorted (sortable) by recency, that nodes query to find out what's the latest one.